api_server.py

```
@@ -549,4 +557,12 @@ def add_security_headers(response):
 """
 
 if __name__ == "__main__":
+    # Production safeguard: do not run the development server in production
+    if os.environ.get("FLASK_ENV") == "production":
+        logger.error(
+            "CRITICAL: Attempted to start Flask development server in production."
+        )
+        logger.error("Use a production WSGI server like Gunicorn instead.")
+        sys.exit(1)
+
     app.run(host="0.0.0.0", port=int(os.environ.get("PORT", 5000)))
```

Ruff (0.14.14)

[error] Possible binding to all interfaces

(S104)